### PR TITLE
Tickets/dm 44771: client-side specification load

### PR DESCRIPTION
**Pull request by eacharles**
Implement client-side specification load:

These required implementing client-side commands to get many of the specification related (SpecBlock, ScriptTemplate) by name to be able to implement the upsert logic.

### CODE REVIEW

File: src/lsst/cmservice/client/loaders.py [hunk 1]
```diff
@@ -1,28 +1,373 @@
...
+
+            if script_template_assoc is None:
+                script_template_assoc = self._parent.script_template_association.create(
+                    spec_name=spec_name,
+                    **script_template_config_,
+                )
+                out_dict["ScriptTemplateAssociation"].append(script_template_assoc)
+                continue
+
+            # FIXME
```
Comment (fritzm) [inline src/lsst/cmservice/client/loaders.py:248]: Was this FIXME fixed just above, and if so, can it go?

Comment (fritzm) [reply src/lsst/cmservice/client/loaders.py:248]: Oh, nm, I see this is for the update case, not create.  Maybe spare a few more words for posterity about what the problem was that needs to be fixed?

File: src/lsst/cmservice/client/loaders.py [hunk 2]
```diff
@@ -1,28 +1,373 @@
...
+
+            if spec_block_assoc is None:
+                spec_block_assoc = self._parent.spec_block_association.create(
+                    spec_name=spec_name,
+                    **spec_block_config_,
+                )
+                out_dict["SpecBlockAssocication"].append(spec_block_assoc)
+                continue
+
+            # FIXME
```
Comment (fritzm) [inline src/lsst/cmservice/client/loaders.py:281]: Similar here -- spare a few words for fix what/why?  Think of the children!

File: src/lsst/cmservice/client/wrappers.py [hunk 3]
```diff
@@ -247,23 +247,64 @@ def get_object_by_fullname_function(
...
-    def get_obj_by_fullname(
+    def get_row_by_fullname(
         obj: CMClient,
         fullname: str,
     ) -> response_model_class:
         params = models.FullnameQuery(
             fullname=fullname,
         )
         results = obj.client.get(f"{query}", params=params.model_dump()).json()
+        if "detail" in results:
```
Comment (fritzm) [inline src/lsst/cmservice/client/wrappers.py:261]: Whut? Why? <stratches head>

File: src/lsst/cmservice/client/wrappers.py [hunk 4]
```diff
@@ -247,23 +247,64 @@ def get_object_by_fullname_function(
...
+
+    def get_row_by_name(
+        obj: CMClient,
+        name: str,
+    ) -> response_model_class | None:
+        params = models.NameQuery(
+            name=name,
+        )
+        results = obj.client.get(f"{query}", params=params.model_dump()).json()
+        if "detail" in results:
```
Comment (fritzm) [inline src/lsst/cmservice/client/wrappers.py:300]: Same puzzle as above.  NO DETAILS FOR YOU!

File: src/lsst/cmservice/client/wrappers.py (hunk 3 above)
Comment (eacharles) [reply src/lsst/cmservice/client/wrappers.py:261]: Cuz that's the easiest way to tell it bombed...  If you have a better way, I'm happy to use it.

File: src/lsst/cmservice/client/wrappers.py (hunk 4 above)
Comment (eacharles) [reply src/lsst/cmservice/client/wrappers.py:300]: Is there a better way to see if `obj.client.get(f"{query}", params=params.model_dump())` is failing?   It doesn't throw or return None like a normal person..

**Review by fritzm — APPROVED** [review]
Thanks!  Couple minor comments left which amount to just requests for a few clarifying comments in the code.

File: src/lsst/cmservice/client/loaders.py (hunk 1 above)
Comment (eacharles) [reply src/lsst/cmservice/client/loaders.py:248]: I was too lazy to redo the association stuff for the update case, mostly b/c we should probably just ditch the whole SpecBlockAssociation and ScriptTemplateAssociation thing.  It is doing the same thing as the spec_aliases, and that fits better into how we do everything else.

**Pull request merged**